web_server.py

```
@@ -0,0 +1,263 @@
+                with open(full_path, 'rb') as f:
+                    self.wfile.write(f.read())
+                return
+        
+        # Default handler for other files
+        super().do_GET()
+
+def run_server(port=12000):
+    """Run the APK hosting server"""
+    handler = APKServerHandler
+    
+    with socketserver.TCPServer(("0.0.0.0", port), handler) as httpd:
```

Ruff (0.13.2)

Possible binding to all interfaces

(S104)